### DISCUSSION TITLE
Display multiple lines in indicator plot

### DISCUSSION THREAD

**Discussion by botradingblog1** [kernc/backtesting.py#665] [Q&A] [answered]
Hi guys,

I was wondering if there's a way to plot multiple lines in the indicator graphs, e.g. MACD and MADC signal line?

Currently, I implement the MACD and MADC signal as two separate indicators.

Any guidance greatly appreciated.

```
def MACD(df_in, fast, slow, signal):
    df = df_in.copy()

    df.ta.macd(close='Close', fast=fast, slow=slow, signal=signal, append=True)
    macd_series = df[f'MACD_{fast}_{slow}_{signal}']
    return macd_series

def MACD_SIGNAL(df_in, fast, slow, signal):
    df = df_in.copy()

    df.ta.macd(close='Close', fast=fast, slow=slow, signal=signal, append=True)
    macd_signal_series = df[f'MACDs_{fast}_{slow}_{signal}']
    return macd_signal_series

Strategy():
    def init(self):
        # Compute indicators
        self.macd1 = self.I(MACD, self.data.df, self.macd_1_fast, self.macd_1_slow, self.macd_1_signal)
        self.macd1_signal = self.I(MACD_SIGNAL, self.data.df, self.macd_1_fast, self.macd_1_slow, self.macd_1_signal)

```

**Comment by kernc** [comment, marked as answer]
Quoting the API reference for [`Strategy.I`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.I):
> `func` is a function that returns the indicator **array(s)** of same length as [Strategy.data](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.data).

You can simply return a sequence of arrays to have them contained in the same indicator plot:
```py
self.macd = self.I(lambda *args: (MACD(*args), MACD_SIGNAL(*args)),
                   self.data.df, self.macd_1_fast, self.macd_1_slow, self.macd_1_signal)

**Reply by botradingblog1** [reply]
Great, thanks very much!
